CONTRIBUTING.md

```diff
@@ -0,0 +1,231 @@
+
+If you need to defer code to a helper function because you require reusing it or otherwise think it's a cleaner approach, put it where you need it: Either as a subfunction in your backend method in the `synadm/api` module or in the frontend function in `synadm/cli/yourcommand`. If you think you've wrote a function that is very generic and might be useful to other `synadm/api` methods as well, put it next to the helpers in the `ApiRequest` class and tell us about it in a PR-comment.
+
+### Logging
+
+Simply use `self.log.warn()` (or `.error/.debug/.info`) within the `synadm/api` module.
+
+From a command's frontend code in the `synadm/cmd` package, logging is available via the `helper` object that is available to every subcommand function, for example use `helper.log.warn()`.
+
+### Code Documentation
+
+We provide an auto-generated [Python package documentation](https://synadm.readthedocs.io/en/latest/index_modules.html) generated with the help of the _autodoc_ feature of the _Sphinx documentation generator_. This requires that code is documented through properly formatted docstrings of functions, classes and methods.
+
+We use the [google docstring format](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings).
+
+_Not all parts of `synadm` are populated with such docstrings yet but we require it for newly written code._
+
+### Code Style
+
+We try to follow Python's [PEP 8](https://www.python.org/dev/peps/pep-0008/) guidelines as good as possible. We have an
+automatic linter in place in our CI pipeline which runs whenever code is pushed to a branch. You'll notice a check going green or red at the very bottom when viewing your open PR. The check is called "Lint with flake8". If after a `git push` you notice the check gone red, click on it and you'll see details of what the linter detected you should correct.
+
+`flake8` can be used directly on your code as well. Install it to your dev-environment:
+
+```
+pip install flake8
+```
+
+Run with:
+
+`flake8 the_file_being_worked_on.py`
+
+or leave out the filename to run it on all files and subdirectories.
+
+Some more style and coding requirements:
+
+- Pythonism: `synadm` is a Python project and as such tries to use some of the very handy features of the language. Some  of those might not be familiar to you since often we realize that contributors are professional programmers used to other languages or even are admins who are rather at home on the shell than in Python code. That is certainly all fine and we are trying to support you with refactoring your code to become more pythonic.
+- That said, we would still like `synadm` to stay approachable to Python beginners and would encourage contributors to try not to overcomplicate things.
+- We require a line length of 79 characters in Python code.
```

Comment:
(Off-topicish) Why specifically 79? Because that means the code is at most 79 characters long, which is 1 off from 80.

Reply:
PEP8 requirese 79. Not true? Or am I misunderstanding your comment?

Reply:
> PEP8 requirese 79. Not true?

I don't know. I don't feel like checking either.

> Or am I misunderstanding your comment?

Probably yes. My question is why 79 characters at most for 1 line instead of 80 characters?

Reply:
Sorry my bad, I understand now I also recall now and I chekcked: Yes, PEP 8 suggestion definitely is 79: https://peps.python.org/pep-0008/

I think I had this misunderstanding once myself and thought that 80 is the pep 8 suggestion. I realized it once I started to use editors and linters that have a "default pep 8" understanding, that it's 79. I then decided that it's best to follow this suggestion of 79 instead of having to reconfigure such tools as well as notify contributors that a non-pep8 default of 80 is used.